### PR TITLE
KAFKA-4019: Update log cleaner to handle max message size of topics

### PR DESCRIPTION
**Pull request by rajinisivaram**
Grow read and write buffers of cleaner up to the maximum message size of the log being cleaned if the topic has larger max message size than the default config of the broker.

### CODE REVIEW

**Comment by omkreddy** [comment]
LGTM.

File: core/src/test/scala/unit/kafka/log/CleanerTest.scala [hunk 1]
```diff
@@ -544,9 +574,12 @@ class CleanerTest extends JUnitSuite {
   
   def key(id: Int) = ByteBuffer.wrap(id.toString.getBytes)
   
-  def message(key: Int, value: Int) = 
+  def message(key: Int, value: Int) : ByteBufferMessageSet =
```
Comment (ijuma) [inline core/src/test/scala/unit/kafka/log/CleanerTest.scala]: Nitpick: there should not be a space before the colon.

Comment (rajinisivaram) [reply core/src/test/scala/unit/kafka/log/CleanerTest.scala]: @ijuma Thanks for the review. Removed the space.

**Comment by becketqin** [comment]
The patch LGTM. This patch is targeted for 0.10.1.0. ping @ijuma

**Comment by hachikuji** [comment]
Thanks for the patch! LGTM.

**Comment by hachikuji** [comment]
Merged to trunk and 0.10.1 with a minor doc tweak.

**Pull request closed without merging**